src/pykeen/models/base.py

```diff
@@ -71,6 +73,110 @@ def _extend_batch(
     return hrt_batch
 
 
+def get_novelty_mask(
+    mapped_triples: MappedTriples,
+    query_ids: np.ndarray,
+    col: int,
+    other_col_ids: Tuple[int, int],
+) -> np.ndarray:
+    r"""Calculate for each query ID whether it is novel.
+
+    In particular, computes:
+
+    .. math ::
+        q \notin \{t[col] in T \mid t[\neg col] = p\}
+
+    for each q in query_ids where :math:`\neg col` denotes all columns but `col`, and `p` equals `other_col_ids`.
+
+    :param mapped_triples: shape: (num_triples, 3), dtype: long
+        The mapped triples (i.e. ID-based).
+    :param query_ids: shape: (num_queries,), dtype: long
+        The query IDs. Are assumed to be unique (i.e. without duplicates).
+    :param col:
+        The column to which the query IDs correspond.
+    :param other_col_ids:
+        Fixed IDs for the other columns.
+
+    :return: shape: (num_queries,), dtype: bool
+        A boolean mask indicating whether the ID does not correspond to a known triple.
+    """
+    other_cols = sorted(set(range(mapped_triples.shape[1])).difference({col}))
+    other_col_ids = torch.tensor(data=other_col_ids, dtype=torch.long, device=mapped_triples.device)
+    filter_mask = (mapped_triples[:, other_cols] == other_col_ids[None, :]).all(dim=-1)
+    known_ids = mapped_triples[filter_mask, col].unique().cpu().numpy()
+    return np.isin(element=query_ids, test_elements=known_ids, assume_unique=True, invert=True)
+
+
+def get_novelty_all_mask(
```

Comment:
@mberr please implement this function such that it takes a query set of triples and then returns if each is in mapped_triples